doc/ax/ax.txt

```diff
@@ -0,0 +1,2095 @@
+<b>`float@myattribute = 0.0f;`</b> which sets the value of this attribute on the
+geometry to @b `0.0f`.
+@par
+For OpenVDB Points, this kernel is run over every point in an OpenVDB Points
+Grid. In simple pseudo code:
+@code{.unparsed}
+foreach(point in pointgrid)
+    run_ax_kernel(point)
+done()
+@endcode
+Where @b `pointgrid` is a single Point Data Grid. For OpenVDB volumes the kernel
+is run over each voxel in a provided grid:
+@code{.unparsed}
+foreach(grid in grids)
+    foreach(voxel IN grid)
+        run_ax_kernel(voxel)
+    done()
+done()
+@endcode
+@par
+Where @b `grids` can comprise of any number of OpenVDB volumes. In this program,
+only a floating point grid with the name @b `myattribute` will be updated. Each
+voxel in this grid will have it's value compared in the same way as the points
+example given above.
+
+
+@subsection axexecutionxcontext OpenVDB Execution Context
+@par
+AX programs are designed to run over different types of geometry, specifically
+OpenVDB points and OpenVDB volumes, the latter of which is the set of all
+numerical OpenVDB grids. Programs must be compiled separately for points and
+volumes, or twice to support both (see the @ref vdbaxbinary "vdb_ax binary").
+This is known as compiling for a target <b>Execution Context</b>. There are two
+main considerations to make when switching between execution contexts:
+    - Certain @ref axfunctions "functions" may not be available under a given
+    context. Some natively supported functions are designed to interface
+    directly with geometry attributes, and a further subset of these are tailored
+    specifically for point attributes @b or voxel values, not both.
+    - Whilst the AX grammar does not change syntactically, @ref axattribaccess
+    "attribute accesses" may be affected. This is due to the fundamental
+    differences of executing over different types of VDBs/geometry.
+@par Execution over Points
+AX kernels compiled for OpenVDB Points run @b individually on <b>each point</b>
+in every <b>active voxel</b> in a points VDB. Attributes that have been
+accessed in the program are provided such that the program has access to all
+data available on the currently processing point. Multiple OpenVDB Point grids
+can be processed by the same AX program, but only a single OpenVDB Points VDB
+can be processed at a time. The default behaviour for AX point programs is to
+process every point which exists in active OpenVDB voxels.
+@par Execution over Volumes
+AX kernels for OpenVDB volumes run individually on every <b>active voxel</b> of
+VDBs which are @b written to (whilst retaining access to all available VDBs).
+Volumes can be thought of as only holding a single "attribute". Whilst points
+hold relevant attributes on the same element in the geometry (assuming no
+neighbour accesses), multiple volume accesses require accessing multiple volume
+primitives. The location of this access is the <b>world space position</b> of
```

Comment:
I think the terms "geometry" and "primitives" in this context could use a more explicit definition

Reply:
I've reworded this section to better describe VDB behaviour